qutip/rcsolve.py

```diff
@@ -0,0 +1,158 @@
+     
+    Returns
+    -------
+    output: Result
+        System evolution.
+    """
+    if options is None:
+        options = Options()
+    output = None
+    start_time = time.time()    
+
+    dot_energy, dot_state = Hsys.eigenstates()
+    deltaE = dot_energy[1] - dot_energy[0]
+    if (Temperature < deltaE/2):
+        warnings.warn("Given temperature might not provide accurate results")
+    gamma = deltaE / (2 * np.pi * wc)
+    wa = 2 * np.pi * gamma *wc #reaction coordinate frequency
+    g = np.sqrt(np.pi * wa * alpha / 2.0) #reaction coordinate coupling
+    nb = (1 / (np.exp(wa/Temperature) -1))
+
+    #Reaction coordinate hamiltonian/operators
+    Nmax = N * 2        #hilbert space 
+    dimensions = dims(Q)
+    a  = tensor(destroy(N), qeye(dimensions[1]))            
+    unit = tensor(qeye(N), qeye(dimensions[1]))
+    Q_exp = tensor(qeye(N), Q)
+    Hsys_exp = tensor(qeye(N),Hsys)
+    return_vals_exp=[tensor(qeye(N), kk) for kk in return_vals]
+
+    na = a.dag() * a    # cavity
+    xa = a.dag() + a
+
+    # decoupled Hamiltonian
+    H0 = wa * a.dag() * a + Hsys_exp
+    #interaction
+    H1 = (g * (a.dag() + a) * Q_exp)
+    H = H0 + H1
+    L=0
+    PsipreEta=0
+    PsipreX=0
+
+    all_energy, all_state = H.eigenstates(sparse=eigen_sparse)
+    Apre = spre((a + a.dag()))
+    Apost = spost(a + a.dag())
+    for j in range(Nmax):
+        for k in range(Nmax):
+            A = xa.matrix_element(all_state[j].dag(), all_state[k])
+            delE = (all_energy[j] - all_energy[k])
+            if np.absolute(A) > 0.0:
+                if abs(delE) > 0.0:
+                    X = (0.5 * np.pi * gamma*(all_energy[j] - all_energy[k])
+                         * (np.cosh((all_energy[j] - all_energy[k]) /
+                            (2 * Temperature))
+                         / (np.sinh((all_energy[j] - all_energy[k]) /
+                            (2 * Temperature)))) * A)
+                    eta = (0.5 * np.pi * gamma *
+                           (all_energy[j] - all_energy[k]) * A)
+                    PsipreX = PsipreX + X * all_state[j]*all_state[k].dag()
+                    PsipreEta = PsipreEta + (eta * all_state[j]
+                                             * all_state[k].dag())
+                else:
+                    X =0.5 * np.pi * gamma * A * 2 * Temperature
+                    PsipreX=PsipreX+X*all_state[j]*all_state[k].dag()
+
+    A = a + a.dag()
+    L = ((-spre(A * PsipreX)) + (sprepost(A, PsipreX))
+         +(sprepost(PsipreX, A)) + (-spost(PsipreX * A))
+         +(spre(A * PsipreEta)) + (sprepost(A, PsipreEta))
+         +(-sprepost(PsipreEta, A)) + (-spost(PsipreEta * A)))           
+
+    #Setup the operators and the Hamiltonian and the master equation 
+    #and solve for time steps in tlist
+    psi0 = (tensor(thermal_dm(N,nb), initial_state))
+    output = mesolve(H, psi0, tlist, [L], return_vals_exp, options=options)
+    end_time = time.time()
+    if calc_time is True:
```

Comment:
This might be better set using the Logger and the debug setting.  The user could do a time calculation outside of the function if they so choose to.  Or they could just go to debug mode if using the Logger. Another option would be to add the calculation time as a property of the Result object.